src/editors/playgroundController.ts

```diff
@@ -492,33 +492,33 @@ export default class PlaygroundController {
     }
 
     try {
-      const progressResult = await vscode.window.withProgress(
+      return await vscode.window.withProgress(
         {
           location: ProgressLocation.Notification,
           title: 'Running MongoDB playground...',
           cancellable: true,
         },
-        async (progress, token) => {
-          token.onCancellationRequested(() => {
-            // If a user clicked the cancel button terminate all playground scripts.
-            this._languageServerController.cancelAll();
-
-            return { result: undefined };
+        async (progress, token): Promise<ShellEvaluateResult> => {
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+          return new Promise(async (resolve, reject) => {
+            token.onCancellationRequested(() => {
+              // If a user clicked the cancel button terminate all playground scripts.
+              this._languageServerController.cancelAll();
+              return resolve({ result: undefined });
+            });
+
+            try {
+              // Run all playground scripts.
```

Comment:
Not related to these changes, since it was already there. Since as we're moving around this code, this comment isn't accurate, it's only running the code that's passed to this function, can we can remove it?
```suggestion
```